content/blogposts/2022/release-3.42

```diff
@@ -0,0 +1,36 @@
+---
+title: "Sourcegraph 3.42 release"
+publishDate: 2022-07-22T10:00-07:00
+description: "Sourcegraph 3.42 introduces UPDATE"
+tags: [blog, release]
+slug: "release/3.42"
+published: false
+heroImage: 
+socialImage: 
+changelogItems:
+
+---
+
+Sourcegraph 3.42 is now available! For this release, we introduced:
+
+## All-new Admin Analytics 
+
+We believe providing a world-class admin-experience is critical to the future of Sourcegraph customers around the world. This starts with empowering administrators to measure and track the value of Sourcegraph within their organization.
+
+Historically, Sourcegraph administrators rely on word-of-mouth feedback from developers, leading to a poor understanding of the value of Sourcegraph within an organization. 
+
+In 3.42, we are releasing a completely redesigned user analytics experience for administrators. These visualizations will make it easier to understand developer engagement, and ultimately better manage user licenses, identify power users, and convey value to their internal leaders. 
+
+## Improved search-based navigation for Python non-locals
+We previously announced we’d [improved our search-based support for Python locals in Sourcegraph 3.40(https://about.sourcegraph.com/blog/release/3.40) by using tree-sitter. We've now extended this improvement for jump-to-definition to more kinds of variables such as class fields.
+
+## Searching by repository descriptions
+We've introduced a new search predicate that enables searching by repository description, for example: `repo:has.description(machine learning)`. Now you can search for interesting repository topics, even if you don't know their names.
+![](https://storage.googleapis.com/sourcegraph-assets/docs/images/code_search/repo-description-search.png)
+
+## New search suggestions
+We've moved to [CodeMirror](https://codemirror.net/) from Monaco for all our search query inputs. We're very excited about the performance and extensibility offered by CodeMirror and look forwaard to shipping all the improvements its flexibility enables. You can switch abck to using Monaco at any time by setting `experimentalFeatures.editor: "monaco"` in your settings menu.
+
+## Lucky search
+We have a new search type available experimentally, called "Lucky search". It does some really cool things to improve the quality of your search experience, but the most obvious one is that it gives smart suggestions for alternate queries, and runs them automatically to make sure they have results. You can enable it by setting `"search.defaultPatternType": "lucky"` in settings.
+![](https://storage.googleapis.com/sourcegraph-assets/docs/images/code_search/lucky-search.png)
```

Comment:
```suggestion
![](https://storage.googleapis.com/sourcegraph-assets/docs/images/code_search/lucky-search.png)

## Code Insights scoped to explicit repository lists load faster and include more historical datapoints

[Gif TODO – need to mock up old and new state next to each other]

Code insights running over an explicit list of repos now behave like code insights running over all repositories. 

They: 
- Backfill 12 historical datapoints (instead of 7)
- Now load 3-10x faster on future page visits thanks to persisting their datapoints
- Better handle large results sets and repositories
- Take daily snapshots of the current state and save it at your defined interval

## Code Insights better supports large monorepos 

Code insights now better supports running insights over large monorepos. Code insights now use our streaming endpoints to avoid being limited by global network timeouts, have improved retry behavior, and send insights running over explicitly-listed individual monorepos to our persistent backend (see the prior paragraph above). 
```